### PR TITLE
Resolved roundoff issue in stddev calc

### PR DESCRIPTION
**Pull request by rppawlo**
Fix round-off issue in the std dev calculation for StackedTimer.

### CODE REVIEW

File: packages/teuchos/comm/src/Teuchos_StackedTimer.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ StackedTimer::printLevel (std::string prefix, int print_level, std::ostream &os,
     if ( options.output_minmax && active_[i]>1) {
       os << " {min="<<min_[i]<<", max="<<max_[i];
       if (active_[i]>1)
-        os<<", std dev="<<sqrt((sum_sq_[i]-sum_[i]*sum_[i]/active_[i])/(active_[i]-1));
+        os<<", std dev="<<sqrt(std::max<double>(sum_sq_[i]-sum_[i]*sum_[i]/active_[i],0.0)/(active_[i]-1));
```
Comment (mhoemmen) [inline packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: yeah this is kinda not the right way to do a one-pass standard deviation.  See Higham, "Accuracy and Stability of Numerical Algorithms," 2nd ed., first chapter

Comment (mhoemmen) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: That reference gives a one-pass mean + variance formula that can never give you a negative variance.

Comment (bathmatt) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: @mhoemmen , we don't have that reference.  Is that algorithm like this??

```
variance(samples):
  M := 0
  S := 0
  for k from 1 to N:
    x := samples[k]
    oldM := M
    M := M + (x-M)/k
    S := S + (x-M)*(x-oldM)
  return S/(N-1)

```

Comment (mhoemmen) [reply packages/teuchos/comm/src/Teuchos_StackedTimer.cpp]: @bathmatt Here's what Higham says, for the mean and variance of `x_1, ..., x_N`:
```
M = x_1
Q = 0
for k in 2 .. N:
    M_old = M
    M = M_old + (x_k - M_old) / k
    Q = Q + (k-1) * (x_k - M_old)^2 / k
```
with variance `Q_N / (N-1)`.  The idea is that `Q_k = \sum_i^k (x_i - M_k)^2`, with `M_k` the running mean `M_k = \frac{1}{k} \sum_i^k x_i`.

**Pull request merged**